source/cellular_at_core.c

```diff
@@ -74,22 +74,21 @@ static uint8_t _charToNibble( char c );
 static void validateString( const char * pString,
                             CellularATStringValidationResult_t * pStringValidationResult )
 {
-    size_t stringLength = 0U;
+    const char * pStringTerminated = NULL;
 
-    /* The strnlen() function returns strlen(s), if that is less than maxlen,
-     * or maxlen if there is no null terminating ('\0') among the first
-     * maxlen characters pointed to by s.
+    /* Validate the string length. If the string length is longer than expected, return
+     * error to stop further processing.
      *
      * stringLength == CELLULAR_AT_MAX_STRING_SIZE is valid because it means that
      * ( CELLULAR_AT_MAX_STRING_SIZE + 1 ) character is null terminating
      * character.*/
-    stringLength = strnlen( pString, CELLULAR_AT_MAX_STRING_SIZE + 1U );
+    pStringTerminated = memchr( pString, '\0', ( CELLULAR_AT_MAX_STRING_SIZE + 1U ) );
```

Comment:
It should not be `+1` with memchr, right? 

```
pStringTerminated = memchr( pString, '\0', CELLULAR_AT_MAX_STRING_SIZE );
```

Reply:
CELLULAR_AT_MAX_STRING_SIZE exclude the NULL character. The comment also help to explain the definition of CELLULAR_AT_MAX_STRING_SIZE.

Reply:
The original design defines CELLULAR_AT_MAX_STRING_SIZE excluding the NULL terminated string.
The longest valid string with length CELLULAR_AT_MAX_STRING_SIZE has '/0' at ( CELLULAR_AT_MAX_STRING_SIZE + 1 ).